src/core/allow-bind.c

```diff
@@ -0,0 +1,188 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#if HAVE_LIBBPF
+#include <bpf/bpf.h>
+#endif
+
+#include "allow-bind.h"
+#include "fd-util.h"
+#include "unit.h"
+
+#if BPF_FRAMEWORK
+/* libbpf, clang and llc compile time dependencies are satisfied */
+#include "bpf/allow_bind/allow-bind-skel.h"
+
+static struct allow_bind_bpf *allow_bind_bpf_free(struct allow_bind_bpf *obj) {
+        if (obj && libbpf_get_error(obj) == 0)
+                (void) allow_bind_bpf__destroy(obj);
+
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct allow_bind_bpf *, allow_bind_bpf_free);
+
+static int can_link_bpf_program(struct bpf_program *prog) {
```

Comment:
What about moving this function to shared place, it could be useful for others as well?

Reply:
@mauriciovasquezbernal

moved: https://github.com/systemd/systemd/pull/17655/files#diff-1f1c18fee968948e02d072a30a51d603bb08a98399bb15ea98c4e0d71d86a669